Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -191,6 +191,18 @@ bool MediaPlayerPrivateGStreamerMSE::doSeek(const MediaTime& position, float rat
 
     // Notify MediaSource and have new frames enqueued (when they're available).
     m_mediaSource->seekToTime(m_seekTime);
+
+    auto player = m_player.get();
+    if (player && !player->isVideoPlayer() && m_audioSink) {
+        gboolean audioSinkPerformsAsyncStateChanges;
+        g_object_get(m_audioSink.get(), "async", &audioSinkPerformsAsyncStateChanges, nullptr);
+        if (!audioSinkPerformsAsyncStateChanges) {
+            // If audio-only pipeline's sink is not performing async state changes
+            // we must simulate preroll right away as otherwise nothing will trigger it.
+            asyncStateChangeDone();
```

Comment:
This makes sense, but we should rename the `asyncStateChangeDone()` to `didPreroll()` so that its name reflects its true purpose. This should be a very simple refactor.